pkg/controllers/provisioning/scheduling/topology.go

```diff
@@ -141,13 +141,19 @@ func (t *Topology) countMatchingPods(ctx context.Context, topologyGroup *Topolog
 
 func TopologyListOptions(namespace string, constraint *v1.TopologySpreadConstraint) *client.ListOptions {
 	selector := labels.Everything()
-	for key, value := range constraint.LabelSelector.MatchLabels {
-		requirement, _ := labels.NewRequirement(key, selection.Equals, []string{value})
-		selector = selector.Add(*requirement)
-	}
-	for _, expression := range constraint.LabelSelector.MatchExpressions {
-		requirement, _ := labels.NewRequirement(expression.Key, selection.Operator(expression.Operator), expression.Values)
-		selector = selector.Add(*requirement)
+	if constraint != nil && constraint.LabelSelector != nil {
```

Comment:
IIUC if LabelSelector is nil, this is considered an invalid configuration. How were you able to even create a pod in the API Server without a label selector? 

You also don't need to check if constraint is nil, since this originates from a `for := range`.